lib/widgets/selection.dart

```diff
@@ -1672,6 +1680,53 @@ class _AddToPlaylistSelectionAction extends StatelessWidget {
   }
 }
 
+/// Action that removes a song from playlist.
+class _FavoriteSelectionAction extends StatefulWidget {
+  const _FavoriteSelectionAction({ Key? key }) : super(key: key);
+
+  @override
+  State<_FavoriteSelectionAction> createState() => _FavoriteSelectionActionState();
+}
+
+class _FavoriteSelectionActionState extends State<_FavoriteSelectionAction> {
+  bool active = false;
+
+  void _handleTap(BuildContext context, ContentSelectionController controller) {
+    final contentTuple = ContentUtils.selectionPack(controller.data);
+    FavoritesControl.instance.setFavorite(
+      contentTuple: contentTuple,
+      value: contentTuple.any((el) => !el.isFavorite),
+    );
+    controller.close();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = getl10n(context);
+    final controller = ContentSelectionController._of(context);
+    final theme = ThemeControl.instance.theme;
+    return _ActionBuilder(
+      controller: controller,
+      shown: () => true,
+      builder: (context, child) {
+        if (controller.inSelection) {
+          active = controller.data.any((el) => !el.data.isFavorite);
+        }
+        return EmergeAnimation(
+          animation: controller.animation,
+          child: HeartButton(
+            tooltip: l10n.addToFavorites,
```

Comment:
There is a TODO comment in the translation files mentioning that this is currently unused. The comment can now be deleted.